File: wagerproof-ios-native/Wagerproof/Features/Analytics/Components/PitcherRegressionCard.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels

/// One starting pitcher regression candidate. Same card for both groups —
/// the negative/positive split lives in the section's group labels; the
/// accent + pill color is driven by per-row severity.
struct PitcherRegressionCard: View {
    let pitcher: MLBPitcherRegression

    var body: some View {
        let sev = Regression.severityColor(pitcher.severity)
        RegressionAccentRow(color: sev) {
            VStack(alignment: .leading, spacing: 10) {
                HStack {
                    VStack(alignment: .leading, spacing: 1) {
                        (Text(pitcher.pitcherName)
                            .font(.system(size: 14, weight: .bold))
                            .foregroundStyle(Color.appTextPrimary)
                         + Text("  \(pitcher.teamName)")
                            .font(.system(size: 12))
                            .foregroundStyle(Color.appTextSecondary))
                            .lineLimit(1)
                        if let opponent = pitcher.opponent {
                            Text("vs \(opponent)")
                                .font(.system(size: 11))
                                .foregroundStyle(Color.appTextSecondary)
                                .lineLimit(1)
                        }
                    }
                    Spacer()
                    RegressionPill(text: pitcher.severity.uppercased(), color: sev)
                }

                HStack(spacing: 10) {
                    RegressionStat(label: "ERA", value: String(format: "%.2f", pitcher.era))
                    RegressionStat(label: "xFIP", value: String(format: "%.2f", pitcher.xfip))
                    RegressionStat(
                        label: "GAP",
                        value: Regression.signed(pitcher.eraMinusXfip, decimals: 2),
                        color: gapColor(pitcher.eraMinusXfip)
                    )
                    RegressionStat(label: "xwOBA", value: pitcher.xwoba.map { String(format: "%.3f", $0) } ?? "-")
                }
                HStack(spacing: 10) {
                    RegressionStat(label: "WHIP", value: pitcher.whip.map { String(format: "%.2f", $0) } ?? "-")
                    RegressionStat(label: "K%", value: pitcher.kPct.map { String(format: "%.1f%%", $0) } ?? "-")
                    RegressionStat(label: "BB%", value: pitcher.bbPct.map { String(format: "%.1f%%", $0) } ?? "-")
                    RegressionStat(
                        label: "xFIP L3",
                        value: pitcher.trendXfip.map { Regression.signed($0, decimals: 2) } ?? "-",
                        color: pitcher.trendXfip.flatMap(trendColor)
                    )
                }
            }
        }
    }

    private func gapColor(_ gap: Double) -> Color? {
        if gap > 0.5 { return Regression.lossRed }
        if gap < -0.5 { return Regression.winGreen }
        return nil
    }

    // Rising xFIP trend = degrading pitcher → red.
    private func trendColor(_ value: Double) -> Color? {
        if value > 0.3 { return Regression.lossRed }
        if value < -0.3 { return Regression.winGreen }
        return nil
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/analytics/PitcherRegressionCard.kt
package com.wagerproof.app.features.analytics

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.MLBPitcherRegression
import java.util.Locale

/**
 * One starting pitcher regression candidate. The negative/positive split lives
 * in the section's group labels; accent + pill color is driven by per-row
 * severity. Port of iOS `PitcherRegressionCard`.
 */
@Composable
fun PitcherRegressionCard(pitcher: MLBPitcherRegression, modifier: Modifier = Modifier) {
    val sev = Regression.severityColor(pitcher.severity)
    RegressionAccentRow(color = sev, modifier = modifier) {
        Column(Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(10.dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Column(Modifier.weight(1f)) {
                    Row {
                        Text(
                            text = pitcher.pitcherName,
                            fontSize = 14.sp,
                            fontWeight = FontWeight.Bold,
                            color = AppColors.appTextPrimary,
                            maxLines = 1,
                            overflow = TextOverflow.Ellipsis,
                        )
                        Text(
                            text = "  ${pitcher.teamName}",
                            fontSize = 12.sp,
                            color = AppColors.appTextSecondary,
                            maxLines = 1,
                            overflow = TextOverflow.Ellipsis,
                        )
                    }
                    if (pitcher.opponent != null) {
                        Text(
                            text = "vs ${pitcher.opponent}",
                            fontSize = 11.sp,
                            color = AppColors.appTextSecondary,
                            maxLines = 1,
                            overflow = TextOverflow.Ellipsis,
                        )
                    }
                }
                RegressionPill(pitcher.severity.uppercase(), sev)
            }

            Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                RegressionStat("ERA", String.format(Locale.US, "%.2f", pitcher.era), Modifier.weight(1f))
                RegressionStat("xFIP", String.format(Locale.US, "%.2f", pitcher.xfip), Modifier.weight(1f))
                RegressionStat("GAP", Regression.signed(pitcher.eraMinusXfip, 2), Modifier.weight(1f), gapColor(pitcher.eraMinusXfip))
                RegressionStat("xwOBA", pitcher.xwoba?.let { String.format(Locale.US, "%.3f", it) } ?: "-", Modifier.weight(1f))
            }
            Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                RegressionStat("WHIP", pitcher.whip?.let { String.format(Locale.US, "%.2f", it) } ?: "-", Modifier.weight(1f))
                RegressionStat("K%", pitcher.kPct?.let { String.format(Locale.US, "%.1f%%", it) } ?: "-", Modifier.weight(1f))
                RegressionStat("BB%", pitcher.bbPct?.let { String.format(Locale.US, "%.1f%%", it) } ?: "-", Modifier.weight(1f))
                RegressionStat(
                    "xFIP L3",
                    pitcher.trendXfip?.let { Regression.signed(it, 2) } ?: "-",
                    Modifier.weight(1f),
                    pitcher.trendXfip?.let { trendColor(it) },
                )
            }
        }
    }
}

private fun gapColor(gap: Double): Color? = when {
    gap > 0.5 -> Regression.lossRed
    gap < -0.5 -> Regression.winGreen
    else -> null
}

// Rising xFIP trend = degrading pitcher → red.
private fun trendColor(value: Double): Color? = when {
    value > 0.3 -> Regression.lossRed
    value < -0.3 -> Regression.winGreen
    else -> null
}
